docs/source/howto/run_containerized_codes.rst

```diff
@@ -0,0 +1,151 @@
+
+.. important::
+
+    Running code in the container usually require mapping the current directory (where the input files locate) to the working directory in the container.
+    Current directory is specified in the job script as ``$PWD``.
+    The computer setup to run containerized code **MUST** set ``use_double_quotes`` to ``True`` to escape the command line parameters using double quotes.
+
+There are two type of codes the ``installed`` and ``portable``, which are all supported to be run in the container.
+
+We use `singularity`_ as the containerized technology in this section.
+Singularity is easy to install and designed to supported the scientific computiong in HPC workd.
+There are other similar containerized technologies for HPC such as `sarus`_ and `shifter`_ are supposed to be support by this feature.
+We have Sarus and Singularity tested to run with this containerized code setup.
+
+Installed containerized code
+============================
+
+The typical code setup configure file for for running code installed singularity container is shown below.
+Assuming we want to run the `/bin/sh` provided from the inside the image [`docker://alpine`].
+
+    .. code-block:: yaml
+
+        ---
+        label: add-singularity
+        description: add executed on singularity
+        default_calc_job_plugin: core.arithmetic.add
+        computer: localhost
+        filepath_executable: /bin/sh
+        image_name: docker://alpine
+        engine_command: singularity exec --bind $PWD:$PWD {image_name}
+        prepend_text: ' '
+        append_text: ' '
+
+Create the code by executing:
+
+.. code-block:: console
+
+    $ verdi code create core.code.installed.containerized --config singularity-add.yaml
+
+Then this code can be used the same way as regular setup code.
+With the advantage that there is no need to explicitly install or compile the code on the remote machine.
+It requires the remote machine has Singularity installed (if not ask HPC administrator to do it) that the executable is in the container shipped with the image.
+For the first time running the containerized code, the image will pulled from dockerhub automatically, which may takes a while if the image is large.
+
+Portable containerized code
+===========================
+
+The portable containerized code is very useful for running scripts with certain dependencies required as environment.
+A typical example is if you have python scripts need to run with certain dependencies such as `numpy`, `scipy` etc.
+You want to make AiiDA able to wrapper those code without go to the remote system and configure python environment there.
+With this feature, there is no need to go to remote computer to build a enviroument install required dependencies for running the script, but just build a docker container an use it.
```

Comment:
```suggestion
The :class:`aiida.orm.nodes.data.code.containerized.PortableContainerizedCode` data plugin allows to provide a container image that provides a computing environment.
This is useful when you have a script that can be executed within that environment, as with the portable containerized code the script can then be deployed in the container on any computer that is configured within AiiDA (as long as that computer supports the containerization technology).

Typical example are Python scripts that require a Python environment with specific dependencies such as `numpy` and `scipy`.
With the ``PortableContainerizedCode``, you can simply configure the container to use and the script to run, and it can then be deployed on any ``Computer``.
```